src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs

```diff
@@ -182,6 +184,62 @@ internal Rect GetVisibleBounds()
             return new Rect(topLeft, bottomRight);
         }
 
+        internal void SaveWorkspaceAsImage(string path)
+        {
+            var initialized = false;
+            var bounds = new Rect();
+
+            var dragCanvas = WpfUtilities.ChildOfType<DragCanvas>(this);
+            var childrenCount = VisualTreeHelper.GetChildrenCount(dragCanvas);
+            for (int index = 0; index < childrenCount; ++index)
+            {
+                var child = VisualTreeHelper.GetChild(dragCanvas, index);
+                var firstChild = VisualTreeHelper.GetChild(child, 0);
+                if ((!(firstChild is NodeView)) && (!(firstChild is NoteView)) && (!(firstChild is AnnotationView)))
+                    continue;
+
+                var childBounds = VisualTreeHelper.GetDescendantBounds(child as Visual);
+                childBounds.X = (double)(child as Visual).GetValue(Canvas.LeftProperty);
+                childBounds.Y = (double)(child as Visual).GetValue(Canvas.TopProperty);
+
+                if (initialized)
+                    bounds.Union(childBounds);
+                else
+                {
+                    initialized = true;
+                    bounds = childBounds;
+                }
+            }
+
+            // Add padding to the edge and make them multiples of two (pad 10px on each side).
+            bounds.Width = 20 + ((((int)Math.Ceiling(bounds.Width)) + 1) & ~0x01);
```

Comment:
I was counting on that `else` statement to handle it, do you have other concerns in mind?